essentials/src/test/java/org/dmfs/jems/function/elementary/PairingFunctionTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.function.elementary;
+
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * Unit test for {@link PairingFunction}
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PairingFunctionTest
+{
+
+    @Test
+    public void test()
+    {
+        LeftHand leftHand = new LeftHand();
+        RightHand rightHand = new RightHand();
+
+        PairingFunction<LeftHand, RightHand> pairingFunction = new PairingFunction<>();
+        assertThat(pairingFunction.value(leftHand, rightHand).left(), sameInstance(leftHand));
+        assertThat(pairingFunction.value(leftHand, rightHand).right(), sameInstance(rightHand));
+
+        assertThat(PairingFunction.<LeftHand, RightHand>instance().value(leftHand, rightHand).left(), sameInstance(leftHand));
+        assertThat(PairingFunction.<LeftHand, RightHand>instance().value(leftHand, rightHand).right(), sameInstance(rightHand));
+    }
+
+
+    private static final class LeftHand
```

Comment:
Why did you create these static classes? Wouldn't it be sufficient to create to `new Object()`? Does it add to the test if the values are of a specific type? Same goes for `ValuePairTest`.

Reply:
I think it can add to the correctness guarantees to the test if actual, different types are used at the place of generic types. To 'test' that part as well. Not that important when there is only one type parameter in the scope of the test, but even then it may add to the readability. (That's why I created that `Contract` class in contentpal testing).

So what I mean by correctness is that we could mess up type parameters as well in theory, but if everything is Object in the test, we wouldn't notice.
Maybe we could have GenericType, GenericType1, GenericType2 'dummy' types available for this purpose. Or something shorter.